### PR TITLE
Added capability to use a reified inline function for instantiating a logger

### PR DESCRIPTION
**Pull request by severn-everett**
Addressing issue #179

### CODE REVIEW

**Comment by oshai** [comment]
note there are conflicts with previous PR

File: src/jvmTest/kotlin/mu/KotlinLoggingTest.kt [hunk 1]
```diff
@@ -1,18 +1,20 @@
 package mu
 
+import mu.KotlinLogging.logger
```
Comment (oshai) [inline src/jvmTest/kotlin/mu/KotlinLoggingTest.kt:3]: I think it makes things more confusing

File: src/jvmTest/kotlin/mu/KotlinLoggingTest.kt [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
 import org.junit.Assert.assertEquals
 import org.junit.Test
 import org.slf4j.LoggerFactory
 
-private val logger = KotlinLogging.logger { }
-private val loggerFromSlf4j = KotlinLogging.logger(LoggerFactory.getLogger("mu.slf4jLogger"))
+private val logger = logger { }
+private val loggerFromSlf4j = logger(LoggerFactory.getLogger("mu.slf4jLogger"))
 private val loggerFromSlf4jExtension = LoggerFactory.getLogger("mu.slf4jLoggerExtension").toKLogger()
```
Comment (oshai) [inline src/jvmTest/kotlin/mu/KotlinLoggingTest.kt:11]: can we have a test also when logger() is defined outside of the class?

Comment (severn-everett) [reply src/jvmTest/kotlin/mu/KotlinLoggingTest.kt:11]: I don't think it's possible to use an reified inline extension function at the top level - the compiler only recognizes the other three public `KotlinLogging.logger()` functions as valid. This declaration works:
```
private val inlineLogger = KotlinLogging.logger()
```
However, this is declaring a logger for `mu.KotlinLogging`, not for the class containing the top-level variables.

File: src/jvmTest/kotlin/mu/KotlinLoggingTest.kt (hunk 1 above)
Comment (severn-everett) [reply src/jvmTest/kotlin/mu/KotlinLoggingTest.kt:3]: The function has to be imported directly in order for the reified inline function to be executed properly. If it were called using the object name (i.e. `val inlineLoggerInClass = KotlinLogging.logger()`) then the logger will be instantiated for `mu.KotlinLogging`, not for `mu.ForKotlinLoggingTest`.

**Comment by oshai** [comment]
I think that given the limitations of the api shown in the PR, it's better not to merge it.

It's easy to use the new api in a way that will not work as expected.

WDYT?

**Comment by severn-everett** [comment]
I agree, also because it's not really going to be worthwhile in JS because `qualifiedName` isn't allowed in KotlinJS.

**Pull request closed without merging**